src/node_data.cpp

```diff
@@ -256,7 +256,7 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+  m_map.push_back({ &key, &value });
```

Comment:
agree, or, to be consistent with line below it:
push_back(kv_pair(&key, &value));